x86-64 build failure is an existing issue with darwin toolchain on 15.4:

```
[217/1028] Validate texture
FAILED: generated/shaders/render/TextureShader.h generated/shaders/render/TextureShader.cpp generated/shaders/render/TextureShaderConstants.h /tmp/nix-build-vengi-tools-0.0.35.d
rv-0/source/build/generated/shaders/render/TextureShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/TextureShader.cpp /tmp/nix-build-vengi-t
ools-0.0.35.drv-0/source/build/generated/shaders/render/TextureShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/texture --constantstemplate 
/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shader
tool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-0
.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53381]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[218/1028] Validate color
FAILED: generated/shaders/render/ColorShader.h generated/shaders/render/ColorShader.cpp generated/shaders/render/ColorShaderConstants.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/s
ource/build/generated/shaders/render/ColorShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ColorShader.cpp /tmp/nix-build-vengi-tools-0.0.3
5.drv-0/source/build/generated/shaders/render/ColorShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/color --constantstemplate /t
mp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shaderto
ol/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-0.0
.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53165]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[219/1028] Validate combine2
FAILED: generated/shaders/render/Combine2Shader.h generated/shaders/render/Combine2Shader.cpp generated/shaders/render/Combine2ShaderConstants.h /tmp/nix-build-vengi-tools-0.0.3
5.drv-0/source/build/generated/shaders/render/Combine2Shader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/Combine2Shader.cpp /tmp/nix-build-ve
ngi-tools-0.0.35.drv-0/source/build/generated/shaders/render/Combine2ShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/combine2 --constantstemplate
 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shade
rtool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-tools-
0.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53240]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[220/1028] Validate convolution
FAILED: generated/shaders/render/ConvolutionShader.h generated/shaders/render/ConvolutionShader.cpp generated/shaders/render/ConvolutionShaderConstants.h /tmp/nix-build-vengi-to
ols-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShader.h /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShader.cpp /t
mp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/ConvolutionShaderConstants.h 
cd /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/src/modules/render && /nix/store/x9390krmrb55ymmxp9xclnyhprbdim49-cmake-3.31.6/bin/cmake -E env APP_HOMEPATH=/tmp/nix-bui
ld-vengi-tools-0.0.35.drv-0/source/build/src/modules/render/ LSAN_OPTIONS=exitcode=0 /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool -I /tmp/nix
-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders --shader /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/modules/render/shaders/convolution --constantstempl
ate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderConstantsTemplate.h.in --headertemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/sh
adertool/ShaderTemplate.h.in --sourcetemplate /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/src/tools/shadertool/ShaderTemplate.cpp.in --buffertemplate /tmp/nix-build-vengi-too
ls-0.0.35.drv-0/source/src/tools/shadertool/UniformBufferTemplate.h.in --sourcedir /tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/generated/shaders/render/
dyld[53309]: Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <68068DCF-9C83-3402-BC24-6446165606F4> /private/tmp/nix-build-vengi-tools-0.0.35.drv-0/source/build/shadertool/vengi-shadertool
  Expected in:     <F0AC1571-049A-3F10-BA96-5C6AD56E68C0> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
Subprocess aborted
[221/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/MaterialColor.cpp.o
[222/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/SurfaceExtractor.cpp.o
[223/1028] Building CXX object src/modules/palette/CMakeFiles/palette.dir/private/PixeloramaPalette.cpp.o
[224/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/Face.cpp.o
[225/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/private/MarchingCubesSurfaceExtractor.cpp.o
[226/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/Mesh.cpp.o
[227/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/private/CubicSurfaceExtractor.cpp.o
[228/1028] Building CXX object src/modules/voxel/CMakeFiles/voxel.dir/MeshState.cpp.o
ninja: build stopped: subcommand failed.

```

--- a/pkgs/applications/graphics/vengi-tools/default.nix
+++ b/pkgs/applications/graphics/vengi-tools/default.nix
@@ -27,9 +27,6 @@
   SDL2,
   SDL2_mixer,
   wayland-protocols,
-  Carbon,
-  CoreServices,
-  OpenCL,
 
   callPackage,
   nixosTests,
@@ -45,6 +42,15 @@ stdenv.mkDerivation (finalAttrs: {
     rev = "v${finalAttrs.version}";
     hash = "sha256-8rGnW+VtqNJYqUqQDp0yOVIQd7w+cq7PIpqqIQPhkbE=";
   };
+
+  prePatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # Disable code signing on macOS
+    substituteInPlace cmake/macros.cmake --replace-fail "codesign" "true"
+    substituteInPlace cmake/system/apple.cmake --replace-fail "if(APPLE)" "if(false)"
+
+    # calls otool -L on /usr/lib/libSystem.B.dylib and fails because it doesn't exist
+    substituteInPlace cmake/applebundle.cmake --replace-fail 'fixup_bundle("''${TARGET_BUNDLE_DIR}" "" "")' ""
+  '';
 
   nativeBuildInputs = [
     cmake
@@ -73,14 +79,7 @@ stdenv.mkDerivation (finalAttrs: {
       SDL2_mixer
     ]
     ++ lib.optional stdenv.hostPlatform.isLinux wayland-protocols
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
-      Carbon
-      CoreServices
-      OpenCL
-    ]
     ++ lib.optional (!stdenv.hostPlatform.isDarwin) opencl-headers;
-
-  cmakeFlags = lib.optional stdenv.hostPlatform.isDarwin "-DCORESERVICES_LIB=${CoreServices}";
 
   # error: "The plain signature for target_link_libraries has already been used"
   doCheck = false;
@@ -89,17 +88,24 @@ stdenv.mkDerivation (finalAttrs: {
     gtest
   ];
 
-  # Set the data directory for each executable. We cannot set it at build time
-  # with the PKGDATADIR cmake variable because each executable needs a specific
-  # one.
-  # This is not needed on darwin, since on that platform data files are saved
-  # in *.app/Contents/Resources/ too, and are picked up automatically.
-  postInstall = lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
-    for prog in $out/bin/*; do
-      wrapProgram "$prog" \
-        --set CORE_PATH $out/share/$(basename "$prog")/
-    done
-  '';
+  postInstall =
+    if stdenv.hostPlatform.isDarwin then
+      ''
+        mkdir -p $out/Applications
+        mv $out/*.app $out/Applications/
+      ''
+    else
+      # Set the data directory for each executable. We cannot set it at build time
+      # with the PKGDATADIR cmake variable because each executable needs a specific
+      # one.
+      # This is not needed on darwin, since on that platform data files are saved
+      # in *.app/Contents/Resources/ too, and are picked up automatically.
+      ''
+        for prog in $out/bin/*; do
+          wrapProgram "$prog" \
+            --set CORE_PATH $out/share/$(basename "$prog")/
+        done
+      '';
 
   passthru.tests = {
     voxconvert-roundtrip = callPackage ./test-voxconvert-roundtrip.nix { };
@@ -124,6 +130,5 @@ stdenv.mkDerivation (finalAttrs: {
     ];
     maintainers = with maintainers; [ fgaz ];
     platforms = platforms.all;
-    broken = stdenv.hostPlatform.isDarwin;
   };
 })

--- a/pkgs/applications/graphics/vengi-tools/default.nix
+++ b/pkgs/applications/graphics/vengi-tools/default.nix
@@ -93,6 +93,9 @@ stdenv.mkDerivation (finalAttrs: {
       ''
         mkdir -p $out/Applications
         mv $out/*.app $out/Applications/
+
+        mkdir -p $out/bin
+        ln -s $out/Applications/vengi-voxconvert.app/Contents/MacOS/vengi-voxconvert $out/bin/vengi-voxconvert
       ''
     else
       # Set the data directory for each executable. We cannot set it at build time

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -16453,9 +16453,7 @@ with pkgs;
 
   vdirsyncer = with python3Packages; toPythonApplication vdirsyncer;
 
-  vengi-tools = darwin.apple_sdk_11_0.callPackage ../applications/graphics/vengi-tools {
-    inherit (darwin.apple_sdk_11_0.frameworks) Carbon CoreServices OpenCL;
-  };
+  vengi-tools = callPackage ../applications/graphics/vengi-tools { };
 
   veusz = libsForQt5.callPackage ../applications/graphics/veusz { };
 

